src/sentry/integrations/pagerduty/utils.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import absolute_import
+
+import logging
+import six
+
+from sentry.incidents.models import IncidentStatus
+from sentry.integrations.metric_alerts import incident_attachment_info
+from sentry.models import PagerDutyService
+from sentry.shared_integrations.exceptions import ApiError
+
+from .client import PagerDutyClient
+
+logger = logging.getLogger("sentry.integrations.pagerduty")
+
+
+def build_incident_attachment(incident, integration_key, metric_value=None):
+    data = incident_attachment_info(incident, metric_value)
+    if incident.status == IncidentStatus.CRITICAL.value:
+        severity = "critical"
+    elif incident.status == IncidentStatus.WARNING.value:
+        severity = "warning"
+    elif incident.status == IncidentStatus.CLOSED.value:
+        severity = "info"
+
+    footer_text = "Sentry Incident | {}".format(data["ts"].strftime("%b %d"))
+
+    return {
+        "routing_key": integration_key,
+        "event_action": "trigger"
+        if incident.status in [IncidentStatus.WARNING.value, IncidentStatus.CRITICAL.value]
+        else "resolve",
+        "dedup_key": "incident_{}".format(incident.identifier),
+        "payload": {
+            "summary": data["text"],
+            "severity": severity,
+            "source": incident.identifier,
+            "custom_details": footer_text,
+        },
+        "links": [{"href": data["title_link"], "text": data["title"]}],
+    }
+
+
+def send_incident_alert_notification(action, incident, metric_value):
+    integration = action.integration
+    service = PagerDutyService.objects.get(organization_integration__integration=integration)
```

Comment:
@ceorourke This doesn't seem right to me. Shouldn't we be querying the `PagerDutyService` object for this particular incident/rule? This is going to blow up if you have more than one `PagerDutyService` for that `integration` (`get` can only return exactly one value or else it will error). You should definitely add a unit test for this case.